core/hydromet_stratified.py

```diff
@@ -557,4 +748,119 @@ def plot_max_potential_retention_cond_runoff(GEV_parameters_E: np.ndarray, PMP:
     ax.set_ylabel('$p_S(S | Q)$')
     ax.set_title('Conditional Max Potential Retention Distribution')
     plt.tight_layout()
-    return None
\ No newline at end of file
+    return None
+
+def precip_to_runoff_h1(hydro_events:np.ndarray,nrcs_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame, df_weights_rainfall: pd.DataFrame, CN: int, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, nrcs temporal distribution, CN and applies the CN reduction method to
+    obtain a runoff curve for each recurrence interval
+    """
+    runoff_distros1 = {}
+    prep_curves = pd.DataFrame(columns = hydro_events.astype(float))
+    for evnt in hydro_events:
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, evnt, display_print)
+        dist_df['precip'] = dist_df['ratio']*precip_data['Median'].loc[evnt]
+        s = S_24hr(CN)
+        ia = IA_24hr(s)
+        runoff_distros1[evnt] = excess_precip(dist_df,ia, s)
```

Comment:
It looks like you are taking advantage of the mutability of the dataframe, by making changes within a function, but not actually returning the dataframe. You want to be more explicit here. Set the function equal to dist_df, and then make that equal to runoff_distros1[evnt].

Reply:
Agreed. runoff_distros1 was not a required variable. the function is now equal to dist_df